scene/2d/camera_2d.cpp

```diff
@@ -41,8 +41,10 @@ void Camera2D::_update_scroll() {
 	}
 
 	if (Engine::get_singleton()->is_editor_hint()) {
-		update(); //will just be drawn
-		return;
+		update();
+		// Only set viewport transform when not bound to the main viewport
+		if (get_viewport() == get_tree()->get_edited_scene_root()->get_viewport())
+			return;
 	}
```

Comment:
Needs braces around the `return;` as in your 4.x PR, that's the coding convention even for one liners.